### PR TITLE
resync ceph-iscsi-gw with old upstream

### PR DESCRIPTION
**Pull request by leseb**
Taken from https://github.com/pcuzner/ceph-iscsi-ansible/tree/tcmu-fixes

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
todo:

- [ ] code import

- [ ] update: generate_group_vars_sample
- [ ] put variables in ceph-defaults
- [ ] add CI tests

- [ ] update `site.yml.sample`

File: roles/ceph-iscsi-gw/tasks/main.yml [hunk 1]
```diff
@@ -1 +1,58 @@
 ---
+- name: os compatibility check (rhel 7.3)
+  assert:
+    that:
+      - "ansible_distribution == 'RedHat'"
+      - "ansible_distribution_version == '7.3'"
```
Comment (dillaman) [inline roles/ceph-iscsi-gw/tasks/main.yml]: Needs to be bumped to 7.4 (or greater)

**Comment by dillaman** [comment]
@leseb Question: why is all the package installation logic being moved to the ```ceph-common``` role? It seems like that would make ```ceph-common``` pretty heavy-weight as opposed to keeping role-specific things within the role itself.

File: roles/ceph-iscsi-gw/tasks/main.yml (hunk 1 above)
Comment (leseb) [inline roles/ceph-iscsi-gw/tasks/main.yml]: Sure.

**Comment by leseb** [comment]
@dillaman the install logic moved to `ceph-common` simply because this is where we install all the packages. This is mainly for consistency purpose.

**Comment by leseb** [comment]
@dillaman actually I haven't moved anything yet, but this is the plan...

**Comment by leseb** [comment]
@dillaman @pcuzner I almost done, I just need to add some tests, do you have an example file/variable to setup gateways? Thanks!

**Comment by dillaman** [comment]
@leseb I don't have anything -- but are these what you are looking for? [1][2]

[1] https://github.com/pcuzner/ceph-iscsi-ansible/blob/master/ceph-iscsi-gw.yml
[2] https://github.com/pcuzner/ceph-iscsi-ansible/blob/master/group_vars/ceph-iscsi-gw.sample

**Comment by leseb** [comment]
@dillaman I'll give this https://github.com/pcuzner/ceph-iscsi-ansible/blob/master/group_vars/ceph-iscsi-gw.sample a try

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-centos7_cluster

**Comment by leseb** [comment]
@ktdreyer are they any community packages for `ceph-iscsi-config`?

**Comment by dillaman** [comment]
@leseb Only developer packages [1]

[1] https://shaman.ceph.com/repos/ceph-iscsi-config/

**Comment by leseb** [comment]
@dillaman thanks!

**Comment by leseb** [comment]
We need a CentOS 7.4 OS for the CI to verify the scenario...

**Comment by dillaman** [comment]
@leseb CentOS 7.4 has been out for a couple weeks -- is that just a note to yourself or do we need someone to take action to update some Jenkins slave somewhere?

**Comment by leseb** [comment]
@dillaman we use this box for vagrant: https://app.vagrantup.com/centos/boxes/7 and it seems to still be running 7.3. Looks like we might have to wait for 1708 :/

**Comment by dillaman** [comment]
@leseb cool -- so nothing for us to do on the backend until that image is released in a week or so (hopefully)

**Comment by leseb** [comment]
@dillaman yes hopefully :-).

**Comment by leseb** [comment]
@dillaman just ran a quick test by removing the 7.3 check, it seems that `tcmu-runner` doesn't seem to be available on shaman.

**Comment by dillaman** [comment]
The tcmu-runner, ceph-iscsi-config, etc seem to be deleted after a week or so. Guess I'll need to ping someone about keeping at least the latest master branch version of these repos indefinitely.

**Comment by leseb** [comment]
@dillaman ok, let's go ahead with this anyway then, so QA can start testing. I disabled the CI on this one.

**Comment by dillaman** [comment]
@leseb awesome -- thank you!

**Comment by dillaman** [comment]
Related PR to keep the latest builds in shaman: https://github.com/ceph/chacra/pull/238

**Comment by leseb** [comment]
@dillaman thanks! Once this merged I'll have another PR to validate some of the installation.

**Pull request merged**